service/src/main/kotlin/fi/espoo/evaka/messaging/message/MessageControllerCitizen.kt

```diff
@@ -125,22 +126,28 @@ class MessageControllerCitizen(
     ): List<UUID> {
         Audit.MessagingCitizenSendMessage.log()
         val accountId = requireMessageAccountAccess(db, user)
-        return db.transaction { tx ->
-            val contentId = tx.insertMessageContent(body.content, accountId)
-            val sentAt = HelsinkiDateTime.now()
-            val threadId = tx.insertThread(MessageType.MESSAGE, body.title)
-            val messageId =
-                tx.insertMessage(
-                    contentId = contentId,
-                    threadId = threadId,
-                    sender = accountId,
-                    sentAt = sentAt,
-                    recipientNames = body.recipients.map { it.name }
-                )
-            body.recipients.map {
-                tx.insertRecipients(setOf(it.id), messageId)
-                threadId
+        val validReceivers = db.read { it.getCitizenReceivers(accountId) }
+        val allReceiversValid = !body.recipients.map { validReceivers.contains(it) }.contains(false)
```

Comment:
I believe this could be written as `body.recipients.all { validReceivers.contains(it) }`